FileCloner/ViewModels/MainPageViewModel.TreeViewGenerator.cs

```diff
@@ -1,58 +1,76 @@
-    /// </summary>
-    private void TreeGenerator(string filePath)
-    {
-        try
-        {
-            // Clear any existing nodes in the tree and reset counters
-            Tree.Clear();
-            ResetCounts();
-            if (filePath == Constants.OutputFilePath)
-            {
-                RootGenerator(filePath);
-                return;
-            }
-
-            // Generate input file representing the structure of the root directory
-            _fileExplorerServiceProvider.GenerateInputFile(RootDirectoryPath);
-
-            // Parse the input file and create tree nodes
-            RootGenerator(Constants.InputFilePath);
-        }
-        catch (Exception e)
-        {
-            // Show error if tree generation fails
-            MessageBox.Show(e.Message);
-        }
-    }
-
-    /// <summary>
-    /// Parses the JSON file representing directory structure and generates the root node.
-    /// </summary>
-    private void RootGenerator(string filePath)
+﻿/******************************************************************************
+ * Filename    = MainPageViewModel.TreeViewGenerator.cs
+ *
+ * Author(s)      = Sai Hemanth Reddy & Sarath A
+ * 
+ * Project     = FileCloner
+ *
+ * Description = Generates the tree view of the directory selected, by recursively 
+ *               populating node and its children with data.
+ *****************************************************************************/
+
+using FileCloner.Models;
+using System.IO;
+using System.Text.Json;
+using System.Windows;
+
+namespace FileCloner.ViewModels;
+
+partial class MainPageViewModel : ViewModelBase
+{
+    /// <summary>
+    /// Generates the initial tree structure of the directory specified in RootDirectoryPath.
+    /// </summary>
+    private void TreeGenerator(string filePath)
     {
         try
         {
-            string jsonContent = File.ReadAllText(filePath);
-            Dictionary<string, JsonElement>? rootDictionary = JsonSerializer.Deserialize<Dictionary<string, JsonElement>>(jsonContent);
+            // Clear any existing nodes in the tree and reset counters
+            Tree.Clear();
+            ResetCounts();
+            if (filePath == Constants.OutputFilePath)
+            {
+                RootGenerator(filePath);
+                return;
+            }
+
+            // Generate input file representing the structure of the root directory
+            _fileExplorerServiceProvider.GenerateInputFile(RootDirectoryPath);
+
+            // Parse the input file and create tree nodes
+            RootGenerator(Constants.InputFilePath);
+        }
+        catch (Exception e)
+        {
+            // Show error if tree generation fails
+            MessageBox.Show(e.Message);
+        }
+    }
 
+    /// <summary>
+    /// Parses the JSON file representing directory structure and generates the root node.
+    /// </summary>
+    private void RootGenerator(string filePath)

```

Comment:
could you check this commit & see if the functionality is same, last night the tree view broke after clicking on summarise so I made changes & pushed this commit ?

 https://github.com/dshreddy/SoftwareEngineering2024/commit/efddfd2e8c90cd658cd3ca214505bfd9fc132c62

Reply:
Seems to be working fine when tested in mine.